app/collins/util/security/MixedAuthenticationProvider.scala

```diff
@@ -1,21 +1,70 @@
 package collins.util.security
 
+import java.util.concurrent.TimeUnit
+
 import collins.models.User
 
+import com.google.common.cache.CacheBuilder
+import com.google.common.cache.CacheLoader
+import com.google.common.cache.LoadingCache
+
 /**
  * Exception encountered during authentication phase
  */
 class AuthenticationException(msg: String) extends Exception(msg)
 
 /**
- * Provides authentication by a variety of methods
+ * Provides authentication by a variety of methods and caching logic (implements decorator pattern)
```

Comment:
Auth caching is an orthogonal concern and should not seep out to the rest of the code (i.e there should be no calls to `tryAuthCache` outside of this class)